### DISCUSSION TITLE
External Code Running with PyBricks Hub

### DISCUSSION THREAD

**Discussion by DylanBPY** [pybricks/support#1789] [Q&A] [answered]
Hello all!
I am working on a program which will need to be able to run a Python script on the brick without using the code.pybricks.com UI. I just need the ability to pass a Python script (or something similar) and have it run on the Spike or Inventor brick running PyBricks.

Also, a not needed thing, but something that would be nice is being able to view connections to the ports (and possibly values?).
Any help on this matter is greatly appreciated!

Note, it would be nice if this was all possible through Javascript/Typescript (I am using React).

- Dylan

**Comment by dlech** [comment]
We have `pybricksdev` for this.

https://github.com/pybricks/pybricksdev?tab=readme-ov-file#command-line-tool

**Reply by DylanBPY** [reply]
This would work fine, but I don't think I can do this through the web. The site will be hosted on a GitHub page, this brings alot of limitations to what can be done sadly. Ill do some more looking tomorrow, but if you have any ideas on how I could do this on  a GitHub Page, it would be very much appreciated!

- Dylan

**Reply by dlech** [reply, marked as answer]
You can borrow from Pybricks Code, e.g. https://github.com/pybricks/pybricks-code/blob/master/src/ble/sagas.ts

**Reply by DylanBPY** [reply]
Thanks for that! 
Ill test this out when I get the opportunity.

- Dylan

**Reply by DylanBPY** [reply]
Follow up here:
https://github.com/orgs/pybricks/discussions/2066
